src/v2/stack_aggregator.py

```diff
@@ -246,22 +247,130 @@ def _get_package_details_with_vulnerabilities(self) -> List[Dict[str, object]]:
+                    vuln_before_ignoring = \
+                        len(package_detail.public_vulnerabilities) + \
+                        len(package_detail.private_vulnerabilities)
+
+                    # If list of vulnerability IDs are provided for current package
+                    # then ignore selected vulnerabilities else ignore all
+                    if packages_to_ignore[package_detail.name]:
+                        public_vulnerabilities = [
+                            vuln for vuln in package_detail.public_vulnerabilities
+                            if vuln.id not in all_vulnerabilities_to_ignore]
+                        private_vulnerabilities = [
+                            vuln for vuln in package_detail.private_vulnerabilities
+                            if vuln.id not in all_vulnerabilities_to_ignore]
+
+                        # Reassign new vulnerabilities after ignoring selected IDs.
+                        package_detail.public_vulnerabilities = public_vulnerabilities
+                        package_detail.private_vulnerabilities = private_vulnerabilities
+
+                        # Number of vulnerabilities after ignoring.
+                        vuln_after_ignoring = \
+                            len(package_detail.public_vulnerabilities) + \
+                            len(package_detail.private_vulnerabilities)
+                    else:
+                        # Ignore all vulnerabilities.
+                        ignore_all_transitives = True
+                        package_detail.public_vulnerabilities = []
+                        package_detail.private_vulnerabilities = []
+
+                        # Set 0 as all vulnerabilities are ignored
+                        vuln_after_ignoring = 0
+
+                    # Get number of vulnerabilities ignored in current package.
+                    ignored_vulnerability_count = vuln_before_ignoring - vuln_after_ignoring
             else:
                 continue  # pragma: no cover
             transitive_details = []
+
             for transitive in transitives:
                 if transitive in self._normalized_packages.dependency_graph:
                     continue
                 transitive_detail = self._normalized_package_details.get(transitive)
+
                 if _has_vulnerability(transitive_detail):
                     transitive_detail = transitive_detail.copy()
+
+                    # Number of transitive vulnerabilities before ignoring.
+                    trans_vuln_before_ignoring = \
+                        len(transitive_detail.public_vulnerabilities) + \
+                        len(transitive_detail.private_vulnerabilities)
+
+                    # User asked to ignore all vulnerabilities of direct dependency
+                    # hence all transitive vulnerabilities are also ignored.
+                    if ignore_all_transitives:
+                        ignored_trans_vulnerability_count = trans_vuln_before_ignoring
```

Comment:
@jparsai I think this should be adding on to `ignored_trans_vulnerability_count` instead of overwriting it.